Tools/CISupport/ews-build/steps.py

```diff
@@ -4938,6 +4957,27 @@ def getResultSummary(self):
         return super().getResultSummary()
 
 
+class ZipFile(shell.ShellCommandNewStyle):
+    name = 'zip-file'
+    descriptionDone = name
+    haltOnFailure = False
```

Comment:
Just to mention, haltOnFailure is false here, but failure of this step will cause subsequent UploadFileToS3() to fail which has haltOnFailure=True.

Reply:
Yes, I would like to address this in a later change. I'm aware that failing to upload a log will fail the build. We need to make `UploadFileToS3` `haltOnFailure` only when uploading a build archive, but not when uploading a log.